packages/nvidia_nat_langchain/src/nat/plugins/langchain/llm.py

```
@@ -366,3 +367,211 @@ async def _agenerate(self,
+    from langchain_core.messages import SystemMessage
+    from langchain_core.outputs import ChatGeneration
+    from langchain_core.outputs import ChatGenerationChunk
+    from langchain_core.outputs import ChatResult
+    
+    try:
+        from huggingface_hub import InferenceClient
+    except ImportError:
+        raise ValueError(
+            "HuggingFace Inference API requires the huggingface_hub package. "
+            "Install it with: pip install huggingface_hub"
+        )
```

Ruff (0.14.14)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)